int32 algorithm
float32 velocity
float32 initial_x
float32 initial_y
float32 desired_x
float32 desired_y
bool simulation
bool reverse
bool choose
---
bool success
string message
